cli/module_graph2.rs

```diff
@@ -845,6 +835,155 @@ impl Graph2 {
     self.modules.contains_key(s)
   }
 
+  pub fn emit(self, options: EmitOptions) -> EmitResult {
+    let mut config = TsConfig::new(json!({
+      "allowJs": true,
+      // TODO(@kitsonk) consider enabling this by default
+      //   see: https://github.com/denoland/deno/issues/7732
+      "emitDecoratorMetadata": false,
+      "esModuleInterop": true,
+      "experimentalDecorators": true,
+      "isolatedModules": true,
+      "jsx": "react",
+      "lib": TypeLib::DenoWindow,
+      "module": "esnext",
+      "strict": true,
+      "target": "esnext",
+    }));
+    let opts = match options.bundle_type {
```

Comment:
In relation to splitting APIs for compile/bundle - if you agree to have separate methods for those purposes I suggest to make those "default" configs `lazy_static` defined at the top of the file.